jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -151,7 +151,11 @@ ecma_new_ecma_string_from_utf8 (const lit_utf8_byte_t *string_p, /**< utf-8 stri
                                 lit_utf8_size_t string_size) /**< string size */
 {
   JERRY_ASSERT (string_p != NULL || string_size == 0);
-  JERRY_ASSERT (lit_is_valid_cesu8_string (string_p, string_size));
+  if (!lit_is_valid_cesu8_string (string_p, string_size))
```

Comment:
I'm not sure this is right fix for this issue. IMHO this is the safest way. Should we fix this in `lexer_construct_regexp_object` instead?